diff --git a/mavlink.patched.js b/mavlink.patched.js
index b67e780..be285ff 100644
--- a/mavlink.patched.js
+++ b/mavlink.patched.js
@@ -2869,7 +2869,7 @@ https://mavlink.io/en/services/parameter.html
 */
 mavlink.messages.param_value = function(param_id, param_value, param_type, param_count, param_index) {
 
-    this.format = '<fHH16sB';
+    this.format = '<4BHH16sB';
     this._id = mavlink.MAVLINK_MSG_ID_PARAM_VALUE;
     this.order_map = [3, 0, 4, 1, 2];
     this.crc_extra = 220;
@@ -2907,7 +2907,7 @@ https://mavlink.io/en/services/parameter.html
 */
 mavlink.messages.param_set = function(target_system, target_component, param_id, param_value, param_type) {
 
-    this.format = '<fBB16sB';
+    this.format = '<4BBB16sB';
     this._id = mavlink.MAVLINK_MSG_ID_PARAM_SET;
     this.order_map = [1, 2, 3, 0, 4];
     this.crc_extra = 168;
@@ -10067,8 +10067,8 @@ mavlink.map = {
         11: { format: '<IBB', type: mavlink.messages.set_mode, order_map: [1, 2, 0], crc_extra: 89 },
         20: { format: '<hBB16s', type: mavlink.messages.param_request_read, order_map: [1, 2, 3, 0], crc_extra: 214 },
         21: { format: '<BB', type: mavlink.messages.param_request_list, order_map: [0, 1], crc_extra: 159 },
-        22: { format: '<fHH16sB', type: mavlink.messages.param_value, order_map: [3, 0, 4, 1, 2], crc_extra: 220 },
-        23: { format: '<fBB16sB', type: mavlink.messages.param_set, order_map: [1, 2, 3, 0, 4], crc_extra: 168 },
+        22: { format: '<4BHH16sB', type: mavlink.messages.param_value, order_map: [3, 0, 4, 1, 2], crc_extra: 220 },
+        23: { format: '<4BBB16sB', type: mavlink.messages.param_set, order_map: [1, 2, 3, 0, 4], crc_extra: 168 },
         24: { format: '<QiiiHHHHBBiIIIIH', type: mavlink.messages.gps_raw_int, order_map: [0, 8, 1, 2, 3, 4, 5, 6, 7, 9, 10, 11, 12, 13, 14, 15], crc_extra: 24 },
         25: { format: '<B20s20s20s20s20s', type: mavlink.messages.gps_status, order_map: [0, 1, 2, 3, 4, 5], crc_extra: 23 },
         26: { format: '<Ihhhhhhhhhh', type: mavlink.messages.scaled_imu, order_map: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10], crc_extra: 170 },
